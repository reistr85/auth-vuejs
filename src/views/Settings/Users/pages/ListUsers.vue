<template>
  <div>
    <PageHeader :schema="$schemas.user" />dddd
    <PageContent>
      <DynamicListPage :schema="$schemas.user" :service="$api.users" />
    </PageContent>
  </div>
</template>

<script>
import PageHeader from '@/components/PageHeader';
import PageContent from '@/components/PageContent';
import DynamicListPage from '@/components/dynamics/DynamicListPage';

export default {
  name: 'ListUser',
  components: { PageHeader, PageContent, DynamicListPage }
};
</script>
