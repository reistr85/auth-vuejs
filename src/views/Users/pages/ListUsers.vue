<template>
  <div>
    <PageHeader :schema="schema" />
    <PageContent>
      <DynamicListPage :schema="schema" />
    </PageContent>
  </div>
</template>

<script>
import PageHeader from '@/components/PageHeader';
import PageContent from '@/components/PageContent';
import UserSchema from '../schemas/UserSchema';
import DynamicListPage from '@/components/DynamicListPage';

export default {
  name: 'ListUser',
  components: { PageHeader, PageContent, DynamicListPage },
  props: {},
  data() {
    return {
      schema: UserSchema,
    }
  },
  methods: {}
}
</script>

<style lang="scss" scoped>

</style>