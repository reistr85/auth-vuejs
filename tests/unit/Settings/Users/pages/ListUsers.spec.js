import { shallowMount } from '@vue/test-utils';
import ListUsers from '@/views/Settings/Users/pages/ListUsers';

describe('Tests component ListUsers', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ListUsers);
  });

  it('should contais components PageHeader, PageContent and DynamicListPage', () => {
    expect(wrapper.findComponent({ name: 'PageHeader' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'PageContent' }).exists()).toBeTruthy();
    expect(wrapper.findComponent({ name: 'DynamicListPage' }).exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  })

  it('should exists state schema and service', () => {
    expect(wrapper.vm.schema).toBeTruthy();
    expect(wrapper.vm.service).toBeTruthy();
  })
})