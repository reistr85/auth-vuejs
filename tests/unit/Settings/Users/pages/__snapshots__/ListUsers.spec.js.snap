// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests component ListUsers should contais components PageHeader, PageContent and DynamicListPage 1`] = `
<div>
  <pageheader-stub schema="[object Object]"></pageheader-stub>
  <pagecontent-stub>
    <dynamiclistpage-stub schema="[object Object]" service="[object Object]"></dynamiclistpage-stub>
  </pagecontent-stub>
</div>
`;
